libV2/tasks/IngestShard.js

```diff
@@ -47,16 +56,16 @@ class IngestShardTask extends BaseTask {
                         if (shardAge < checkpointLagMicroseconds) {
                             metricClass = 'utapi.event';
                             records = metrics
-                                .map(m => new UtapiMetric(JSON.parse(m)));
+                                .map(m => this._hydrateEvent(m));
```

Comment:
is it possible to remove the extra function wrap?
```suggestion
                                .map(this._hydrateEvent);
```

Reply:
You can do something like `this._hydrateEvent.bind(this)`, but it's just doing the same thing with a different syntax. Root problem is that js doesn't carry around a reference to the correct `this`, so in `_hydrateEvent` on L#27 without the wrap/bind `this` would not be pointing to an instance of `IngestShardTask`.

Reply:
right... that dang `this`...